### ISSUE TITLE
user_ip delete duplicates failed, causing lots of fun errors trying to create the index in the background

### ISSUE THREAD

**Issue by turt2live** [matrix-org/synapse#4398]
```
2019-01-15 23:15:58.030 CET [18989] ERROR:  canceling autovacuum task
2019-01-15 23:15:58.030 CET [18989] CONTEXT:  automatic vacuum of table "synapse.public.user_ips"
2019-01-15 23:15:58.078 CET [18806] t2botsynapse@synapse ERROR:  could not serialize access due to concurrent update
2019-01-15 23:15:58.078 CET [18806] t2botsynapse@synapse STATEMENT:  DELETE FROM user_ips WHERE user_id = '@telegram_redacted:t2bot.io' AND access_token = 'redacted' AND ip = 'redacted'
```

then a few hundred of these:

```
2019-01-15 23:17:36.451 CET [18919] t2botsynapse@synapse ERROR:  could not create unique index "user_ips_user_token_ip_unique_index"
2019-01-15 23:17:36.451 CET [18919] t2botsynapse@synapse DETAIL:  Key (user_id, access_token, ip)=(@_discord_redacted:t2bot.io, redacted, redacted) is duplicated.
2019-01-15 23:17:36.451 CET [18919] t2botsynapse@synapse STATEMENT:  CREATE UNIQUE INDEX CONCURRENTLY user_ips_user_token_ip_unique_index ON user_ips (user_id, access_token, ip)
```

**Comment by neilisfragile** [comment]
Related #4370

**Comment by richvdh** [comment]
I think we fixed this in #4434.
